models/src/anemoi/models/data_indices/collection.py

```diff
@@ -127,6 +130,81 @@ def todict(self):
     def representer(dumper, data):
         return dumper.represent_scalar(f"!{data.__class__.__name__}", repr(data))
 
+    def _compare_variables(self, ckpt_name_to_index: dict[str, int], data_name_to_index: dict[str, int]) -> None:
```

Comment:
If this becomes public (used by anemoi-training), you should remove the '_' : `_compare_variables` -> `compare_variables`.

Also, you will need to choose between a method `data_indices.compare_variables(name_to_index)` or a function `compare_variables(checkpoint_name_to_index, data_name_to_index)`. 
I would vote for a function as there are not many "self" in this piece of code.